hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1053,12 +1053,24 @@ public AbfsRestOperation read(final String path,
     return op;
   }
 
-  public AbfsRestOperation deletePath(final String path, final boolean recursive, final String continuation,
+  public AbfsRestOperation deletePath(final String path, final boolean recursive,
+                                      final String continuation,
                                       TracingContext tracingContext)
           throws AzureBlobFileSystemException {
     final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
-
     final AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
+
+    if (abfsConfiguration.isPaginatedDeleteEnabled() && recursive) {
+      // Change the x-ms-version to "2023-08-03" if its less than that.
+      if (xMsVersion.compareTo(AUGUST_2023_API_VERSION) < 0) {
+        requestHeaders.removeIf(header -> header.getName().equalsIgnoreCase(X_MS_VERSION));
```

Comment:
This would be kind of O(n) check on each delete invocation. What if we set the right version in `createDefaultHeaders`.

Reply:
+1 on this

Reply:
Taken